techmatters/serverless
# Chi 2023 warm transfer reservation timeout stuck state

## Description

- Bug fix: Warm transfer reservation timeout causes a stuck state

**Problem**: During a warm transfer, if the receiving counsellor doesn’t action the transfer reservation, the reservation will time out, but the form and transfer state will remain locked for the original counsellor.

**Solution**: Added a check to ascertain whether warm voice transfer timed out and updated the task attribute.

### Checklist
- [N/A] New tests added



### Related Issues
Fixes CHI-2023

### Verification steps

- Login to Aselo as two counsellors. I did this using my Okta login (A) and Twilio login (B).
- Call the Aselo Dev number
- A picks up the call, and WARM transfers (consult) to B.
- B ignore the call so that it will time out. The transfer should time out after 2 minutes.
- Verify that the original counsellor gets the task back in the original state (form unlocked, ability to transfer and conference).
